File: 01-pattern-problems/java/Pattern4.java
import java.util.Scanner;
/** 
 * @author Sourabh Ninawe
 */
class Pattern4{
    public void printPattern4(int n){
        for(int i=1;i<=n;i++){
            for(int j=1;j<=i;j++){
                System.out.print(i);
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        while(true){
            System.out.println("Enter the number of lines to print pattern 4 in integer and -1 to exit");
            Scanner sc = new Scanner(System.in);
            int n = sc.nextInt();
            if(n<=0)
                break;
            else{
                Pattern4 p4 = new Pattern4();
                p4.printPattern4(n);
            }
        }
    }
}

File: 01-pattern-problems/python/pattern4.py

# Author: Sourabh Ninawe
class pattern4:
    def print_pattern4_method1(self, n):
        print("pattern4:Method1")
        for i in range(1, n+1):
            for j in range(1, i+1):
                print(i, end='')
            print()

    #shorter implementation
    def print_pattern4_method2(self, n):
        print("pattern4:Method2")
        for i in range(1, n+1):
            print(str(i)*i)

if __name__ == "__main__":
    while True:
        try:
            n = int(input("Enter the number of lines to print pattern 4 (integer, 0 or negative to exit): "))
        except ValueError:
            print("Please enter a valid integer.")
            continue
        if n <= 0:
            break
        else:
            p3 = pattern4()
            p3.print_pattern4_method1(n)
            p3.print_pattern4_method2(n)
